features/exacte_match.feature

```diff
@@ -0,0 +1,147 @@
+  En één object (3) met huisnummer=3
+  En huisletter=A
+  En huisnummertoevoeging=bis
+  En één object (4) met huisnummer=3
+  En huisletter=B
+  En geen huisnummertoevoeging
+  En één object (5) met huisnummer=3
+  En huisletter=B
+  En huisnummertoevoeging=bis
+  Als er wordt gezocht met postcode=1234AA
+  En huisnummer=3 
+  En huisnummertoevoeging=bis
+  En geen huisletter 
+  En exacteMatch=false
+  Dan bevat het resultaat object (3) met huisnummer=3
+  En huisletter=A 
+  En huisnummertoevoeging=bis 
+  En object (5) met huisnummer=3
+  En huisletter=B
+  En huisnummertoevoeging=bis
+	
+  Gegeven een verzameling van 5 objecten met postcode=1234AA 
+  En één object (1) met huisnummer=3
+  En geen huisletter
+  En geen huisnummertoevoeging
+  En één object (2) met huisnummer=3
+  En huisletter=A
+  En geen huisnummertoevoeging
+  En één object (3) met huisnummer=3
+  En huisletter=A
+  En huisnummertoevoeging=bis
+  En één object (4) met huisnummer=3
+  En huisletter=B
+  En geen huisnummertoevoeging
+  En één object (5) met huisnummer=3
+  En huisletter=B
+  En huisnummertoevoeging=bis
+  Als er wordt gezocht met postcode=1234AA
+  En huisnummer=3 
+  En huisnummertoevoeging=bis
+  En geen huisletter 
+  En exacteMatch=true
+  Dan bevat het resultaat geen objecten
+
+Scenario: combinatie van huisnummer, huisletter en/of huisnummertoevoeging komt meerdere keren voor en exacteMatch wordt niet toegepast
+  Gegeven een verzameling van 2 objecten met postcode=1234AA
+  En huisnummer=3 
+  En geen huisletter 
+  En geen huisnummertoevoeging
+  Als er wordt gezocht met postcode=1234AA
+  En huisnummer=3
+  En geen huisnummertoevoeging
+  En geen huisletter 
+  En exacteMatch=false
+  Dan bevat het resultaat beide objecten
+  
+Scenario: combinatie van huisnummer, huisletter en/of huisnummertoevoeging komt meerdere keren voor en exacteMatch wordt toegepast
+  Gegeven een verzameling van 2 objecten met postcode=1234AA
+  En huisnummer=3 
+  En geen huisletter 
+  En geen huisnummertoevoeging
+  Als er wordt gezocht met postcode=1234AA
+  En huisnummer=3
+  En geen huisnummertoevoeging
+  En geen huisletter 
+  En exacteMatch=true
+  Dan bevat het resultaat beide objecten
+  
+Scenario: endpoints waarbij exacteMatch kan worden toegepast kennen geen fuzzy search
+  Gegeven een verzameling van 2 objecten met postcode=1234AA
+  En één object (1) met huisnummer=3 
+  En één object (2) met huisnummer=13
+  Als er wordt gezocht met postcode=1234AA
+  En huisnummer=3
+  En exacteMatch=false
+  Dan bevat het resultaat alleen object (1) met postcode=1234AA 
+  En huisnummer=3
+
+Scenario: endpoints waarbij exacteMatch kan worden toegepast kennen geen fuzzy search
+  Gegeven een object met postcode= 1234AA 
+  En huisnummer=3 
+  En huisletter=B
+  En geen huisnummertoevoeging
+  Als er wordt gezocht met postcode=1234AA
+  En huisnummer=3
+  En huisnummertoevoeging=B
+  En exacteMatch=false
+  Dan bevat het resultaat geen objecten
```

Comment:
zoekcriteria komen exact overeen met eigenschappen van een adres, maar het gevonden adres wordt niet teruggeven omdat het exact matcht? Krijg je dan UITSLUITEND zoekresultaten terug die NIET exact matchen? Maar dan wordt het zoeken heel erg lastig voor een gebruiker. ExacteMatch= false betekent toch dat ik alles wat exact matcht sowieso terug krijg, PLUS OOK alles wat niet exact matcht? 

Ik zou verwachten dat met de exacteMatch = true een subset van het ExacteMatch = false (fuzzy search resultaat) wordt teruggegeven. Niet een ander resultaat. Het kan natuurlijk heel goed dat ik iets heb gemist in de discussie (of niet helemaal fris meer was :-))

Reply:
In deze situatie is er een object met: postcode=1234AA, huisnummer=3, **huisletter**=B en exacteMatch=false.
Er wordt gezocht met: postcode=1234AA, huisnummer=3 en **huisnummertoevoeging**=B en exacteMatch=false.
In de verzameling van objecten is geen object aanwezig met **huisnummertoevoeging**=B (wel met **huisletter**=B, daarom is het resultaat leeg.
Exacte match en fuzzy search werken verschillend en kunnen niet in combinatie (lees: bij dezelfde endpoint) worden gebruikt.

Reply:
Dan zegt geen resultaat, maar tabel bevat wel resultaat